File: Adapter Pattern/1. Minimal/Java/Main.java
public class Main {
    public static void main(String[] args) {
        System.out.println("Adapter Pattern in Java");
        System.out.println("-----------------------");
        System.out.println();
        
        // Using class adapter.
        Target classAdapter = new ClassAdapter();
        classAdapter.request();

        // Using object adapter.
        Adaptee adaptee = new Adaptee();
        Target objectAdapter = new ObjectAdapter(adaptee);
        objectAdapter.request();
    }
}

File: Adapter Pattern/1. Minimal/Python/main.py
from class_adapter import ClassAdapter
from object_adapter import ObjectAdapter
from adaptee import Adaptee

def main():
    print("Adapter Pattern in Python")
    print("-------------------------")
    print()
    
    # Using class adapter.
    classAdapter = ClassAdapter()
    classAdapter.request()

    # Using object adapter.
    adaptee = Adaptee()
    objectAdapter = ObjectAdapter(adaptee)
    objectAdapter.request()

if __name__ == "__main__":
    main()
